File: 342. Power of Four.java
class Solution {
    public boolean check(int n){
        int mask = 0b01010101010101010101010101010101;
        if((mask | n) == mask) return true;

        return false;
    }
    public boolean isPowerOfFour(int n) {
        if(n > 0 && ((n & (n - 1)) == 0) && check(n)) return true;

        return false;
    }
}

File: 342. Power of Four.cpp
class Solution {
public:
    bool check(int n) {
        int mask = 0b01010101010101010101010101010101;
        return ((mask | n) == mask);
    }

    bool isPowerOfFour(int n) {
        return (n > 0 && ((n & (n - 1)) == 0) && check(n));
    }
};
